src/Pagination.js

```diff
@@ -13,44 +13,44 @@ import PageItem, { First, Prev, Ellipsis, Next, Last } from './PageItem';
  * @property {PageItem} Next
  * @property {PageItem} Last
  */
-class Pagination extends React.Component {
-  static propTypes = {
-    /** @default 'pagination' */
-    bsPrefix: PropTypes.string.isRequired,
-
-    /**
-     * Set's the size of all PageItems.
-     *
-     * @type {('sm'|'lg')}
-     */
-    size: PropTypes.string,
-  };
-
-  render() {
-    const { bsPrefix, className, children, size, ...props } = this.props;
-
+const Pagination = React.forwardRef(
+  ({ bsPrefix, className, children, size, ...props }, ref) => {
+    const decoratedBsPrefix = useBootstrapPrefix(bsPrefix, 'pagination');
     return (
       <ul
+        ref={ref}
         {...props}
         className={classNames(
           className,
-          bsPrefix,
-          size && `${bsPrefix}-${size}`,
+          decoratedBsPrefix,
+          size && `${decoratedBsPrefix}-${size}`,
         )}
       >
         {children}
       </ul>
     );
-  }
-}
-
-const DecoratedPagination = createBootstrapComponent(Pagination, 'pagination');
-
-DecoratedPagination.First = First;
-DecoratedPagination.Prev = Prev;
-DecoratedPagination.Ellipsis = Ellipsis;
-DecoratedPagination.Item = PageItem;
-DecoratedPagination.Next = Next;
-DecoratedPagination.Last = Last;
-
-export default DecoratedPagination;
+  },
+);
+
+Pagination.propTypes = {
+  /**
+   * @default 'pagination'
+   * */
+  bsPrefix: PropTypes.string,
+
+  /**
+   * Set's the size of all PageItems.
+   *
+   * @type {('sm'|'lg')}
+   */
+  size: PropTypes.string,
+};
```

Comment:
I think to keep our code base consistent we should define our propTypes *before* we define the component. Some examples include [Accordion](https://github.com/react-bootstrap/react-bootstrap/blob/master/src/Accordion.js), [Alert](https://github.com/react-bootstrap/react-bootstrap/blob/master/src/Alert.js), [Nav](https://github.com/react-bootstrap/react-bootstrap/blob/master/src/Nav.js), etc. I'm not sure if we want to change this convention in the future though